hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterNetworkTopologyServlet.java

```diff
@@ -204,7 +279,41 @@ public void testPrintTopologyNoDatanodesJsonFormat() throws Exception {
     sb.append("\n-- Network Topology -- ");
     String topology = sb.toString();
 
-    // assert node number
+    // Assert node number.
     assertTrue(topology.contains("No DataNodes"));
   }
 }
+
+class RouterServerHelper implements BeforeEachCallback, AfterAllCallback {
```

Comment:
I have a question. Shouldn't RouterServerHelper be extracted from this class and made into a separate utility class?

Reply:
Hi, @slfan1989 . Actually, I'd ever consider this way, but i found each test class have its own initialize logic. For example, some test class may need mkdis or addMountPoints which is unused for other test class. What's your opinions?

Reply:
"I believe the current changes in this PR are acceptable, but it would be better to extract `RouterServerHelper` to avoid potential code redundancy.

Reply:
Maybe we can merge this firstly and i will try to extract common logic in https://github.com/apache/hadoop/pull/7470 ?

Reply:
@KeeProMise Do you have any other suggestions? We can take a look at the details in #7470 first. If you think there are no issues, we can merge this PR anytime. Thanks again!

Reply:
@hfutatzhanghb @slfan1989 LGTM.